src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -1,39 +1,93 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './NewCommentForm.scss';
+import { Comment } from '../../types/comment';
 
-export const NewCommentForm: React.FC = () => (
-  <form className="NewCommentForm">
-    <div className="form-field">
-      <input
-        type="text"
-        name="name"
-        placeholder="Your name"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <input
-        type="text"
-        name="email"
-        placeholder="Your email"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <textarea
-        name="body"
-        placeholder="Type comment here"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <button
-      type="submit"
-      className="NewCommentForm__submit-button button"
+interface NewComment {
+  name: string;
+  email: string;
+  body: string;
+}
+
+type Props = {
+  addNewComment: (args0: Comment) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  addNewComment,
+}) => {
+  const [newComment, setNewComment] = useState<NewComment>({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const handleChange = (propName: string, propValue: string) => {
+    const updatedComment = {
+      ...newComment,
+      [propName]: propValue,
+    };
+
+    setNewComment(updatedComment);
+  };
+
+  const submitNewComment = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const comment = {
+      ...newComment,
+      id: Math.random(),
+      postId: Math.random() + 1,
```

Comment:
what if math.random will random same number twice?

Reply:
change this logic

Reply:
It handles by special function `verifyUniqueCommentId` in `src\components\PostDetails\PostDetails.tsx`

Reply:
but only for `comment.id`